### ISSUE TITLE
Add aria states and properties to `TreeSection` and `TreeItem` components

### ISSUE THREAD

**Issue by dotNomad** [posit-dev/publisher#1436]
The ARIA states and properties on the `TreeSection` and `TreeItem` components are lighter than they should be. 

For example:
- `aria-expanded` should be used on both
- `aria-level` for the hierarchy level of nesting on `TreeItem`s
- `aria-label` for section description for `TreeSection`s
- `aria-hidden` for decorative icons
